src/dotnet/commands/dotnet-cache/Program.cs

```diff
@@ -70,13 +70,20 @@ public static CacheCommand FromArgs(string[] args, string msbuildPath = null)
             {
                 msbuildArgs = new List<string>();
 
-                if (string.IsNullOrEmpty(projectArgument.Value()))
+                if (!projectArguments.HasValue())
                 {
                     throw new InvalidOperationException(LocalizableStrings.SpecifyEntries);
                 }
 
                 msbuildArgs.Add("/t:ComposeCache");
-                msbuildArgs.Add(projectArgument.Value());
+                msbuildArgs.Add(projectArguments.Values[0]);
+
+                projectArguments.Values.RemoveAt(0);
```

Comment:
It's best to not mutate the `projectArguments` variable.  That way if you ever need to inspect them later, you get the full list. Instead you can do something like this:

```
msbuildArgs.Add($"/p:AdditionalProjects={string.Join("%3B", projectArguments.Skip(1).Values)}");
```

Note the `Skip(1)` in there.